xarray/tests/test_datatree.py

```diff
@@ -1732,19 +1733,61 @@ def test_dim_argument(self):
 
 
 class TestOps:
-    @pytest.mark.xfail(reason="arithmetic not implemented yet")
+    def test_unary_op(self):
```

Comment:
Could you enable type checking in all these tests? (Adding `...(self) -> None:`).

I just remembered that I will have to fix some type hints related to the ops... So in case you get too many errors, just leave it like it is and maybe add a comment here to remind me :)

Reply:
Whoops, a minute too late my review, haha.

I guess this is then up to me when I have time to fix the remaining ops typing issues...

Reply:
https://github.com/pydata/xarray/pull/9632